projects/cds/src/profiletag/services/profile-tag.injector.ts

```diff
@@ -28,15 +34,18 @@ export class ProfileTagInjector {
   private w: ProfileTagWindowObject;
   public consentReference = null;
   public profileTagDebug = false;
-  private tracking$: Observable<boolean | NgRouterEvent> = merge(
+  private tracking$: Observable<boolean> = merge(
     this.pageLoaded(),
-    this.consentChanged()
+    this.consentChanged(),
+    this.cartChanged()
   );
+
   constructor(
     private winRef: WindowRef,
     private config: CdsConfig,
     private baseSiteService: BaseSiteService,
     private router: Router,
+    private cartService: CartService,
```

Comment:
This service is starting to have too many dependencies, maybe we can split it somehow. But let's discuss that for the next, _refactoring_ ticket.

Reply:
This will also make testing a bit easier.

Reply:
I agree. I see 3 distinct things: the 'tracking', the 'profiletagevents' and then the code to manage it. I will change it after this PR